code/modules/mob/living/carbon/human/species_types/felinid.dm

```diff
@@ -131,17 +131,31 @@
 /proc/purrbation_apply(mob/living/carbon/human/H, silent = FALSE)
 	if(!ishuman(H) || isfelinid(H))
 		return
-	H.set_species(/datum/species/human/felinid)
-
+	if(ishumanbasic(H))
+		H.set_species(/datum/species/human/felinid)
+	else
+		var/obj/item/organ/ears/cat/kitty_ears = new
+		var/obj/item/organ/tail/cat/kitty_tail = new
+		kitty_ears.Insert(H, TRUE, FALSE) //Gives nonhumans cat tail and ears
+		kitty_tail.Insert(H, TRUE, FALSE)
 	if(!silent)
 		to_chat(H, "Something is nya~t right.")
 		playsound(get_turf(H), 'sound/effects/meow1.ogg', 50, TRUE, -1)
 
 /proc/purrbation_remove(mob/living/carbon/human/H, silent = FALSE)
-	if(!ishuman(H) || !isfelinid(H))
-		return
-
-	H.set_species(/datum/species/human)
-
+	if(isfelinid(H))
+		H.set_species(/datum/species/human)

```

Comment:
is it intentional that purrbation removes people who were originally felinid? seems like there needs to be a check for that somewhere.

Reply:
It was like that before, but I will add that